xdsl/ir.py

```diff
@@ -118,7 +120,17 @@ class SSAValue(ABC):
     uses: set[Use] = field(init=False, default_factory=set, repr=False)
     """All uses of the value."""
 
-    name: str | None = field(init=False, default=None)
+    _name: str | None = field(init=False, default=None)
+
+    @property
+    def name(self) -> str | None:
+        return self._name
+
+    @name.setter
+    def name(self, name: str):
+        if name[-1].isnumeric():
```

Comment:
AFAIK, this is not the only constraint, right?
We should also check that the contents of `name` are alphanumerics, otherwise we would allow `a b`, or `+a`.

Reply:
Good point!

Reply:
What are the exact constraints on a `SSAValue.name`? MLIR docs provide the following regex: `([0-9]+|([A-Za-z_$.-][\w$.-]*))` which I created from the following spec:

```
    suffix-id ::= (digit+ | ((letter|id-punct) (letter|id-punct|digit)*))
    id-punct  ::= [$._-]
```

Do we just check the regex plus the last character not being numeric?

Reply:
I implemented that for now, if you have any feedback let me know!